pkgs/development/compilers/ghc/common-hadrian.nix

```diff
@@ -596,6 +598,15 @@ stdenv.mkDerivation (
           "-j$NIX_BUILD_CORES"
           ${lib.escapeShellArgs hadrianSettings}
         )
+      ''
+      # "native cross", e.g. pkgsStatic & co, should not be treated as "cross" by hadrian / GHC, otherwise
+      # the internal interpreter and iserve are not built, and Template Haskell will not be supported.
```

Comment:
I wonder if instead of this if we can just override the silly logic in hadrian at the moment and force the `internal-interpreter` flags to be enabled in this case. 

I think that should work for all GHC versions as long as the boot GHC version matches the current one. It should also work for GHC>9.12 without that restriction.